File: src/commands/music/jump.ts
import { SlashCommandBuilder } from "@discordjs/builders";
import { CommandInteraction } from "discord.js";
import { MusicHandler } from "../../util/handlers/music";
import { ExtendedClient, Command } from "../../util/structures";

export const command: Command = {
    data: new SlashCommandBuilder()
        .setName("jump")
        .setDescription("Jumps to a song in queue without removing items. Default is jumping to the next song")
        .addNumberOption(option =>
            option
                .setName("index")
                .setDescription('The number in the queue you want to jump to')
        )
        ,
    async execute(interaction: CommandInteraction, client: ExtendedClient) {
        const MH = new MusicHandler(client)

        var index = interaction.options.getNumber('index')

        await MH.skip(interaction, (index === null ? 1 : index), false)
    }
}

File: bot/extensions/commands/music/jump.py
import discord
from discord import app_commands
from discord.ext import commands

from util import Models, MusicHandler


class Jump(commands.Cog):
    def __init__(self, bot: Models.LavaBot) -> None:
        self.bot = bot
        self.music_handler = MusicHandler(bot)

    @app_commands.command(description="Skips next song in queue by default.")
    @app_commands.describe(index="The number in the queue you want to skip to")
    async def jump(self, itr: discord.Interaction, index: int = None):
        await itr.response.defer()
        await self.music_handler.skip(itr, index if index != None else 1, False)

        return


async def setup(bot: Models.LavaBot):
    await bot.add_cog(Jump(bot))
